admin-dev/themes/new-theme/js/components/grid/extension/position-extension.js

```diff
@@ -83,25 +82,26 @@ export default class PositionExtension {
    * @returns {Array}
    * @private
    */
-  getRowsPositions(paginationOffset) {
+  getRowsPositions() {
     const tableData = JSON.parse($.tableDnD.jsonize());
     const rowsData = tableData[`${this.grid.getId()}_grid_table`];
-    const regex = /^row_(\d+)_(\d+)$/;
+    const completeRowsData = [];
 
-    const rowsNb = rowsData.length;
-    const positions = [];
-    let rowData; let
-      i;
-    for (i = 0; i < rowsNb; i += 1) {
-      rowData = regex.exec(rowsData[i]);
-      positions.push({
-        rowId: rowData[1],
-        newPosition: paginationOffset + i,
-        oldPosition: parseInt(rowData[2], 10),
+    let trData;
+
+    // retrieve dragAndDropOffset offset to have all needed data
+    // for positions mapping evolution over time
+    for (let i = 0; i < rowsData.length; i += 1) {
+      trData = this.grid.getContainer()
+        .find(`#${rowsData[i]}`);
+
```

Comment:
I like your idea to format data before computing them, I think you could even perform the regex here, this would lighten the `computeMappingBetweenOldAndNewPositions` algorithm

for example:

```js
    let rowDataParsedData = regex.exec(rowsData[i]);

    completeRowsData.push({
        rowMarker: rowsData[i],
        offset: trData.data('dragAndDropOffset'),
        rowID: rowDataParsedData[1],
        oldPosition: parseInt(rowDataParsedData[2], 10),
    });
```

This way in the compute function you can directly use `rowsData[i].rowId` for example, which would help the readability